docs/dev/pagination-v2.md

```diff
@@ -0,0 +1,173 @@
+# Pagination in v2 Engine
```

Comment:
I think we need to describe/link what the legacy engine is capable of. 
Then a reader will better understand what the Scope section includes.